### PR TITLE
adds leetcode integration

### PR DESCRIPTION
**Pull request by anuragdaksh7**

### CODE REVIEW

**Comment by anuragdaksh7** [comment]
styling in jsx file is temprory it would be shifted to a dedicated scss file soon

File: services/integrations/leetcode/fetchUserContributions.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import axios from "axios";
+
+const leetcodeGraphQLEndpoint = "https://leetcode.com/graphql/"; // Replace with LeetCode's endpoint if they have one
```
Comment (prashant-software-developer) [inline services/integrations/leetcode/fetchUserContributions.js]: use enums for this, create and enum file if not created

File: services/integrations/leetcode/fetchUserContributions.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  for (const day in calendar) {
+    const date = new Date(Number(day + "000"));
+    const mordifiedDate = date.toISOString().split("T")[0];
+    const tmp = { date: mordifiedDate, count: calendar[day] };
+    mordified.push(tmp);
+  }
+
+  // console.log(mordified)
+
+  return mordified;
```
Comment (prashant-software-developer) [inline services/integrations/leetcode/fetchUserContributions.js]: correct return format in [success, failure]

File: services/integrations/leetcode/fetchUserContributions.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    },
+    operationName: "userProfileCalendar",
+  };
+  const response = await axios.post(leetcodeGraphQLEndpoint, graphqlData);
+  const data = await response.data;
+  const calendar = JSON.parse(
+    data.data.matchedUser.userCalendar.submissionCalendar,
+  );
+
+  const mordified = new Array();
```
Comment (prashant-software-developer) [inline services/integrations/leetcode/fetchUserContributions.js]: nameing?? mordified?

File: app/integrationLeetcode/page.jsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import LeetcodeCalendar from "./Calendar/LeetcodeCalendar";
+import fetchData from "@/services/integrations/leetcode/fetchUserContributions";
+
+const page = async () => {
+  const username = "anuragdaksh";
+  const response = await fetchData(username);
```
Comment (prashant-software-developer) [inline app/integrationLeetcode/page.jsx]: get data in [succ, err] format

**Review by prashant-software-developer — CHANGES_REQUESTED** [review]

File: services/integrations/leetcode/fetchUserContributions.js (hunk 3 above)
Comment (anuragdaksh7) [reply services/integrations/leetcode/fetchUserContributions.js]: i was playing around with the api not thinking of naming 😂, sure will change it soon

**Pull request merged**